demo/src/main/java/com/sothree/slidinguppanel/demo/DemoActivity.kt

```diff
@@ -88,38 +104,35 @@ class DemoActivity : AppCompatActivity() {
         // Inflate the menu; this adds items to the action bar if it is present.
         menuInflater.inflate(R.menu.demo, menu)
         val item = menu.findItem(R.id.action_toggle)
-        if (sliding_layout.panelState == PanelState.HIDDEN) {
+        if (sliding_layout.getPanelState() == PanelState.HIDDEN) {
             item.setTitle(R.string.action_show)
         } else {
             item.setTitle(R.string.action_hide)
         }
         return true
     }
 
-    override fun onPrepareOptionsMenu(menu: Menu): Boolean {
-        return super.onPrepareOptionsMenu(menu)
-    }
-
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         when (item.itemId) {
             R.id.action_toggle -> {
-                if (sliding_layout.panelState != PanelState.HIDDEN) {
-                    sliding_layout.panelState = PanelState.HIDDEN
+                if (sliding_layout.getPanelState() != PanelState.HIDDEN) {
```

Comment:
What's the point to move to `getPanelState()`.
Why not using property access ?

I only get one more additional weak warning

<img width="573" alt="image" src="https://github.com/hannesa2/AndroidSlidingUpPanel/assets/3314607/fae29106-00ad-4ca6-83bf-2e5c8fb6b95b">

Reply:
It looks like, that you are no more interested in applying here some more changes.
I will merge it and I'll will revoke some changes from you by myself.
I've no clue, why you removed property access, but I will revoke this

Reply:
Hi @hannesa2 I'm sorry I have two jobs currently, I am absolutely interested in this still, but I simply did not find time yet! I will look at the merge as soon as I can, please note that the variable panelState internally has a different name, this was causing the endless loop bug before, which I described in my first comment in this PR, it's important to understand which setter vs. field access is called by this method. But as I said, I will have a look at this soon. I'll build it in a clean way so that the IDE doesn't need to show warnings anymore